client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -486,17 +496,23 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
 
     // Second, for each worker, try to initialize the endpoint.
     val parallelism = Math.min(Math.max(1, slots.size()), conf.clientRpcMaxParallelism)
-    ThreadUtils.parmap(slots.asScala, "InitWorkerRef", parallelism) { case (workerInfo, _) =>
-      try {
-        workerInfo.endpoint =
-          rpcEnv.setupEndpointRef(RpcAddress.apply(workerInfo.host, workerInfo.rpcPort), WORKER_EP)
-      } catch {
-        case t: Throwable =>
-          logError(s"Init rpc client failed for $shuffleId on $workerInfo during reserve slots.", t)
-          connectFailedWorkers.put(
-            workerInfo,
-            (StatusCode.WORKER_UNKNOWN, System.currentTimeMillis()))
-      }
+
+    ThreadUtils.parmap(slots.asScala.filter(_._1.endpoint == null), "InitWorkerRef", parallelism) {
```

Comment:
Seems we can also pass the shared thread pool here. We can change the name to shared rpc thread pool.